spec/lib/map/security_token/service_spec.rb

```diff
@@ -22,6 +24,50 @@
         end
       end
 
+      context 'token caching' do
+        let(:map_sts_token) { SecureRandom.hex(100) }
+        let(:expected_log_message) { "#{log_prefix} token success" }
+
+        shared_examples 'new token request' do
+          it 'calls for a new token, caches it, and logs a MAP STS token success message with cache: false',
+             vcr: { cassette_name: 'map/security_token_service_200_response' } do
+            expect(Rails.logger).to receive(:info).once.and_call_original
+            expect(Rails.logger).to receive(:info).with(expected_log_message, { application:, icn:, cache: false })
+            expect(Rails.cache).to receive(:write_entry).with(cache_key, anything,
+                                                              hash_including(expires_in: 5.minutes, force: !cache))
+            expect(subject[:access_token]).not_to eq(map_sts_token)
+          end
+        end
+
+        context 'when the "cache" argument is true' do
+          context 'when a token with a matching application & ICN is cached' do
+            before do
+              allow(Rails.cache).to receive(:fetch).with(cache_key, expires_in: 5.minutes,
+                                                                    force: !cache).and_return(cached_token)
+            end
+
+            let(:cached_token) { { access_token: map_sts_token, expiration: 1.hour.from_now.utc.iso8601 } }
+
+            it 'returns the cached token and logs a MAP STS token success message with cache: true' do
+              expect(Rails.logger).to receive(:info).once.and_call_original
```

Comment:
For these logs I would go ahead and expect the initial log, so basically the same line as what comes next but:
`let(:expected_token_request_log_message) { '[MAP][SecurityToken][Service] token request' }'`
`expect(Rails.logger).to receive(:info).with(expected_token_request_log_message, { application:, icn: })`

Reply:
(to be clear each of these would expect two logs, the first one as described here, and then the second one as you already have